### PR TITLE
MAINT - Update pre-commit hooks

### PR DESCRIPTION
**Pull request by trallard**
As I was reviewing some PRs I noticed our pre-commit hooks needed an update and that some files (HTML) were not being linted/formated.

Included in this PR:

- Update pre-commit hooks and configs in `pyproject.toml`
- Run linter on all files
- Updates docs

There are a lot of changes as I ran the linter, so the only files needing a check are `pyproject.toml` and `.pre-commit-config.yaml`

### CODE REVIEW

**Review by Carreau — APPROVED** [review]

**Comment by Carreau** [comment]
Approved, but this might conflict with other so Maybe merge later as it will be easier to rebase this (and rerun the precomit hooks) than other way around.

**Comment by trallard** [comment]
Good shout on the (not) merging. I added a `do-not-merge` label for now

**Comment by Carreau** [comment]
Rebased / fixed conflicts by regenerating the files.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -48,4 +48,11 @@ repos:
...
       - id: prettier
-        types_or: [css, markdown]
+        types_or: [css, markdown, html]
+
+
+  - repo: https://github.com/Riverside-Healthcare/djLint
+    rev: v1.34.1
+    hooks:
+      - id: djlint-jinja
+        types_or: ["jinja"]
```
Comment (gabalafou) [inline .pre-commit-config.yaml:58]:
```suggestion
```

I got rid of Jinja in #42

**Comment by gabalafou** [comment]
> One thing I did not do was delete the `tests/results/*.html`, as it seem the new scripts don't create those, but create pngs with playwrite, should I delete them ?

#41 (merged) deletes the entire test/results directory, so I'm not sure why that's not a merge conflict with main

**Comment by Carreau** [comment]
All looks good, let's merge, I don't think it will conflicts with #33

**Pull request merged**